simulariumio/readers/histogram_plot_reader.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+from typing import Dict, Any
+
+from ..exceptions import MissingDataError
+from .reader import Reader
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class HistogramPlotReader(Reader):
+    def read(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Return an object containing the data shaped for Simularium format
+        """
+
+        if "title" not in data:
+            raise MissingDataError("title")
+        if "xaxis_title" not in data:
+            raise MissingDataError("xaxis_title")
+        if "traces" not in data:
+            raise MissingDataError("traces")
```

Comment:
You should look into Vega / vega-lite / altair for visualization specification: https://vega.github.io/vega-lite/docs/spec.html

Works in python and javascript so easy to convert between the two and may be good for this exact situation. Just a thought though. Don't know enough about your plot specification.